framework/src/executioners/Steady.C

```diff
@@ -37,5 +39,5 @@ Steady::Steady(const InputParameters & parameters) : Transient(parameters)
     return;
   }
 
-  _problem.transient(false);
+  _num_steps = _problem.adaptivity().getSteps() + 1;
```

Comment:
I guess my point is that *system time* is an important parameter even for Steady. I have paid extra attention in our eigenvalue executioners like `NonlinearEigen` on not changing the system time.

Reply:
That's the way it always was with Steady: time increased by `1` every adaptivity step.  It's just the way it is with MOOSE.

Reply:
It is in `Steady` but not in `NonlinearEigen`. If we want to switch to `Eigenvalue` which is derived from `Steady`, we will have to fix this system time. In `Steady`, if I am understanding correctly, the time is used only for outputting, which can be patched up easily like what is in `NonlinearEigen`. It might be more complicated and harder to fix here because time is used as system time. I thought of changing `FEProblem::outputStep(ExecFlagType type)`, to `FEProblem::outputStep(ExecFlagType type, Real pseudo_time)` so that we can have direct control on what time to use for outputting instead of directly using system time.